tests/errors/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+            let _ = client.hello(&1);
+        })
+    }
+
+    #[test]
+    // TODO: Remove the should_panic when this issue is fixed:
+    // https://github.com/stellar/rs-soroban-sdk/issues/642
+    #[should_panic(expected = "Status(ContractError(1))")]
+    fn hello_error_panic() {
+        let e = Env::default();
+        let contract_id = BytesN::from_array(&e, &[0; 32]);
+        e.register_contract(&contract_id, Contract);
+        let client = ContractClient::new(&e, &contract_id);
+
+        let expected_status = Status::from_contract_error(Error::AnError as u32);
+        e.assert_panic_with_status(expected_status, |_| {
+            let _ = client.hello(&2);
+        })
+    }
+
+    #[test]
+    fn hello_error_panic_string() {
+        let e = Env::default();
+        let contract_id = BytesN::from_array(&e, &[0; 32]);
+        e.register_contract(&contract_id, Contract);
+        let client = ContractClient::new(&e, &contract_id);
+
+        let expected_string = "an error";
+        e.assert_panic_with_string(expected_string, |_| {
+            let _ = client.hello(&3);
+        })
+    }
+
+    #[test]
+    fn try_hello_ok() {
+        let e = Env::default();
+        let contract_id = BytesN::from_array(&e, &[0; 32]);
+        e.register_contract(&contract_id, Contract);
+        let client = ContractClient::new(&e, &contract_id);
+
+        let res = client.try_hello(&0);
+        assert_eq!(res, Ok(Ok(symbol!("hello"))));
+    }
+
+    #[test]
+    fn try_hello_error() {
+        let e = Env::default();
+        let contract_id = BytesN::from_array(&e, &[0; 32]);
+        e.register_contract(&contract_id, Contract);
+        let client = ContractClient::new(&e, &contract_id);
+
+        let res = client.try_hello(&1);
+        assert_eq!(res, Err(Ok(Error::AnError)));
+    }
+
+    #[test]
+    // TODO: Remove the should_panic when this issue is fixed:
+    // https://github.com/stellar/rs-soroban-sdk/issues/642
+    #[should_panic(expected = "Status(ContractError(1))")]
+    fn try_hello_error_panic() {
+        let e = Env::default();
+        let contract_id = BytesN::from_array(&e, &[0; 32]);
+        e.register_contract(&contract_id, Contract);
+        let client = ContractClient::new(&e, &contract_id);
+
+        let res = client.try_hello(&2);
+        assert_eq!(res, Err(Ok(Error::AnError)));
+    }
+
+    #[test]
+    // TODO: Remove the should_panic when this issue is fixed:
+    // https://github.com/stellar/rs-soroban-env/issues/430
+    fn try_hello_error_panic_string() {
+        let e = Env::default();
+        let contract_id = BytesN::from_array(&e, &[0; 32]);
+        e.register_contract(&contract_id, Contract);
+        let client = ContractClient::new(&e, &contract_id);
+
+        let res = client.try_hello(&3);
+        assert_eq!(
+            res,
+            Err(Err(Status::from_status(ScStatus::VmError(
+                ScVmErrorCode::TrapUnreachable
+            ),)))
+        );
```

Comment:
@graydon There's some error cases in here that I'm not sure what to do with, would appreciate your thoughts. I've opened two issues about them.
- https://github.com/stellar/rs-soroban-env/issues/430
- https://github.com/stellar/rs-soroban-sdk/issues/642